### PR TITLE
New package promise improvements.

### PR DESCRIPTION
**Pull request by pasinskim**

### CODE REVIEW

**Comment by pasinskim** [comment]
@kacfengine you can take a look at this. This should address https://dev.cfengine.com/issues/7328

**Comment by kacfengine** [comment]
Excellent!

**Comment by kacfengine** [comment]
trigger build

**Comment by kacfengine** [comment]
This broke quite a lot of stuff in The Great Package Test, which surprises me. Did you try that one?

**Comment by pasinskim** [comment]
@kacfengine I will try to take a look at this stuff.

**Comment by pasinskim** [comment]
trigger build

**Comment by pasinskim** [comment]
@kacfengine I forgot to commit this stuff yesterday evening. Build should be started so once is green we can merge.

File: cf-agent/package_module.c [hunk 1]
```diff
@@ -1472,6 +1486,25 @@ PromiseResult HandlePresentPromiseAction(EvalContext *ctx,
...
+            {
+                Log(LOG_LEVEL_VERBOSE,
+                    "Unexpected package architecture received");
+                free(package_info->arch);
+                package_info->arch = NULL;
+            }
+            if (package_info->version)
+            {
+                Log(LOG_LEVEL_VERBOSE,
+                    "Unexpected package version received");
```
Comment (kacfengine) [inline cf-agent/package_module.c]: Suggest to change this to ``Unexpected package version received from package module``, just to make the context clear. Same for architecture.

**Comment by kacfengine** [comment]
I will just change the stuff I mentioned locally. The build is green, so I will merge this.

**Comment by kacfengine** [comment]
Changed to ``Unexpected package architecture received from package module. Ignoring.`` and merged manually. Cherry-picked to 3.7.x as well.

**Pull request closed without merging**